test/utils/GroupCallUtils-test.ts

```diff
@@ -0,0 +1,659 @@
+            room: roomId,
+            user: userId1,
+            event: true,
+            type: "test",
+            skey: userId1,
+            content: {},
+        });
+    };
+
+    const setUpEmptyStateKeyCallEvent = () => {
+        callEvent = mkEvent({
+            room: roomId,
+            user: userId1,
+            event: true,
+            type: "org.matrix.msc3401.call",
+            skey: "",
+            content: {},
+        });
+    };
+
+    const setUpValidCallEvent = () => {
+        callEvent = mkEvent({
+            room: roomId,
+            user: userId1,
+            event: true,
+            type: "org.matrix.msc3401.call",
+            skey: callId,
+            content: {},
+        });
+    };
+
+    beforeEach(() => {
+        client = createTestClient();
+    });
+
+    describe("getGroupCall", () => {
+        describe("for a non-existing room", () => {
+            beforeEach(() => {
+                mocked(client.getRoom).mockReturnValue(null);
+            });
+
+            it("should return null", () => {
+                expect(getGroupCall(client, roomId)).toBeUndefined();
+            });
+        });
+
+        describe("for an existing room", () => {
+            let room: Room;
+
+            beforeEach(() => {
+                room = new Room(roomId, client, client.getUserId());
+                mocked(client.getRoom).mockImplementation((rid: string) => {
+                    return rid === roomId
+                        ? room
+                        : null;
+                });
+            });
+
+            it("should return null if no 'call' state event exist", () => {
+                expect(getGroupCall(client, roomId)).toBeUndefined();
+            });
+
+            describe("with call state events", () => {
+                let callEvent1: MatrixEvent;
+                let callEvent2: MatrixEvent;
+                let callEvent3: MatrixEvent;
+
+                beforeEach(() => {
+                    callEvent1 = mkEvent({
+                        room: roomId,
+                        user: client.getUserId(),
+                        event: true,
+                        type: "org.matrix.msc3401.call",
+                        content: {},
+                        ts: 150,
+                        skey: "call1",
+                    });
+                    room.getLiveTimeline().addEvent(callEvent1, {
+                        toStartOfTimeline: false,
+                    });
+
+                    callEvent2 = mkEvent({
+                        room: roomId,
+                        user: client.getUserId(),
+                        event: true,
+                        type: "org.matrix.msc3401.call",
+                        content: {},
+                        ts: 100,
+                        skey: "call2",
+                    });
+                    room.getLiveTimeline().addEvent(callEvent2, {
+                        toStartOfTimeline: false,
+                    });
```

Comment:
I think this should be `sendStateEvent()`?

Reply:
(applies to the whole file)

Reply:
`sendStateEvent()` doesn't work on the mock client. This is for the test setup.

Reply:
Ah, that's a shame, we should probably get it working in the future

Reply:
Depends on how unitish we want to be in our tests. I don't care about the client here. Instead I want to test the GroupCallUtils with minimal dependencies. One could argue that we should use as many real dependencies as possible to get the highest possible coverage. This is the classic Unit vs Integration discussion :laughing: